src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -320,6 +320,62 @@ public List<JobModel> enumerateJobs(
         .collect(Collectors.toList());
   }
 
+  private List<FlightState> getAccessibleFlights(
+      int offset, int limit, FlightFilter filter, AuthenticatedUserRequest userReq)
+      throws InterruptedException {
+    int start = 0;
+    int end = offset + limit;
+    HashMap<String, List<String>> roleMap = new HashMap<>();
+    ArrayList<FlightState> flightStateList = new ArrayList<>();
+    while (flightStateList.size() < offset + limit) {
+      List<FlightState> filterResults = stairway.getFlights(start, end, filter);
+      if (filterResults.size() == 0) {
+        break;
+      }
+      filterResults.forEach(
+          flightState -> {
+            if (userLaunchedFlight(flightState, userReq)) {
+              flightStateList.add(flightState);
+            } else {
+              FlightMap inputParameters = flightState.getInputParameters();
+              IamResourceType resourceType =
+                  inputParameters.get(
+                      JobMapKeys.IAM_RESOURCE_TYPE.getKeyName(), IamResourceType.class);
+              String resourceId =
+                  inputParameters.get(JobMapKeys.IAM_RESOURCE_ID.getKeyName(), String.class);
+              IamAction action =
+                  inputParameters.get(JobMapKeys.IAM_ACTION.getKeyName(), IamAction.class);
+              if (resourceType != null & resourceId != null && action != null) {
```

Comment:
Any jobs that do not have the resourceType, resourceId and action in the flight parameters will only show up in the enumeration if the user launched that job (or if the user is an admin) so the existing behavior will not change.

Reply:
Got it, thanks for confirming!  And that makes sense that all recent job access expansion is forward-looking only, not just this enumeration.  I think that's reasonable for most of our use cases, but good to be aware of in case someone is looking at older jobs and seeing unexpected behavior.